From 1daea46e8887bff910e63eac77ece2273fb0783d Mon Sep 17 00:00:00 2001
From: Yu-Han Yang <yuhany@google.com>
Date: Tue, 21 Sep 2021 12:29:34 -0700
Subject: [PATCH 62/82] [Q_asb_2021-12] Fix serialization bug in
 GpsNavigationMessage

Bug: 196970023
Test: presubmits passing.
Change-Id: I69f51eb2faac0cf2ee9f7a5f94f7100925f7221c
(cherry picked from commit a1b4297002b927ce0fc795c0ac1d45a6a200b43b)
---
 location/java/android/location/GpsNavigationMessage.java | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/location/java/android/location/GpsNavigationMessage.java b/location/java/android/location/GpsNavigationMessage.java
index 77f011374dab..10cdae2a1da3 100644
--- a/location/java/android/location/GpsNavigationMessage.java
+++ b/location/java/android/location/GpsNavigationMessage.java
@@ -259,12 +259,8 @@ public class GpsNavigationMessage implements Parcelable {
             parcel.readByteArray(data);
             navigationMessage.setData(data);
 
-            if (parcel.dataAvail() >= Integer.SIZE) {
-                int status = parcel.readInt();
-                navigationMessage.setStatus((short) status);
-            } else {
-                navigationMessage.setStatus(STATUS_UNKNOWN);
-            }
+            int status = parcel.readInt();
+            navigationMessage.setStatus((short) status);
 
             return navigationMessage;
         }
-- 
2.35.1

